app/drivers/sensor/cirque_trackpad/cirque_trackpad.c

```diff
@@ -0,0 +1,256 @@
+    k_sem_give(&data->gpio_sem);
+#elif defined(CONFIG_PINNACLE_TRIGGER_GLOBAL_THREAD)
+    k_work_submit(&data->work);
+#endif
+}
+#endif
+
+#define SPI_BUS DT_BUS(DT_DRV_INST(0))
+#define SPI_REG DT_REG_ADDR(DT_DRV_INST(0))
+
+static int pinnacle_init(const struct device *dev) {
+    struct pinnacle_data *data = dev->data;
+    const struct pinnacle_config *config = dev->config;
+
+    pinnacle_write(dev, PINNACLE_STATUS1, 0);   // Clear CC
+    pinnacle_write(dev, PINNACLE_Z_IDLE, 0);    // No Z-Idle packets
+    if (config->sleep_en) {
+        pinnacle_write(dev, PINNACLE_SYS_CFG, PINNACLE_SYS_CFG_EN_SLEEP);
+    }
+    if (config->no_taps) {
+        pinnacle_write(dev, PINNACLE_FEED_CFG2, PINNACLE_FEED_CFG2_DIS_TAP);
+    }
+    uint8_t feed_cfg1 = PINNACLE_FEED_CFG1_EN_FEED;
+    if (config->invert_x) {
+        feed_cfg1 |= PINNACLE_FEED_CFG1_INV_X;
+    }
+    if (config->invert_y) {
+        feed_cfg1 |= PINNACLE_FEED_CFG1_INV_Y;
+    }
+    if (feed_cfg1) {
+        pinnacle_write(dev, PINNACLE_FEED_CFG1, feed_cfg1);
+    }
+
+#ifdef CONFIG_PINNACLE_TRIGGER
+    data->dev = dev;
+    gpio_pin_configure_dt(&config->dr, GPIO_INPUT);
+    gpio_init_callback(&data->gpio_cb, pinnacle_gpio_cb, BIT(config->dr.pin));
+    int ret = gpio_add_callback(config->dr.port, &data->gpio_cb);
+    if (ret < 0) {
+        LOG_ERR("Failed to set DR callback: %d", ret);
+        return -EIO;
+    }
+
+#if defined(CONFIG_PINNACLE_TRIGGER_OWN_THREAD)
+    k_sem_init(&data->gpio_sem, 0, UINT_MAX);
+
+    k_thread_create(&data->thread, data->thread_stack, CONFIG_PINNACLE_THREAD_STACK_SIZE,
+                    (k_thread_entry_t) pinnacle_thread, (void *) dev, 0, NULL,
+                    K_PRIO_COOP(CONFIG_PINNACLE_THREAD_PRIORITY), 0, K_NO_WAIT);
+#elif defined(CONFIG_PINNACLE_TRIGGER_GLOBAL_THREAD)
+    k_work_init(&data->work, pinnacle_work_cb);
+#endif
+    pinnacle_write(dev, PINNACLE_FEED_CFG1, feed_cfg1);
+#endif
+    LOG_WRN("inited");
+    return 0;
+}
+
+static const struct sensor_driver_api pinnacle_driver_api = {
+#if CONFIG_PINNACLE_TRIGGER
+	.trigger_set = pinnacle_trigger_set,
+#endif
+	.sample_fetch = pinnacle_sample_fetch,
+	.channel_get = pinnacle_channel_get,
+    .attr_set = pinnacle_attr_set,
+};
+
+static struct pinnacle_data pinnacle_data;
+static const struct pinnacle_config pinnacle_config = {
+#if DT_INST_ON_BUS(0, i2c)
+	.bus = I2C_DT_SPEC_INST_GET(0),
+#elif DT_INST_ON_BUS(0, spi)
+	.bus = SPI_DT_SPEC_INST_GET(0, SPI_OP_MODE_MASTER | SPI_WORD_SET(8) | SPI_LINES_SINGLE | SPI_TRANSFER_MSB, 0),
+#endif
+    .invert_x = DT_INST_PROP(0, invert_x),
+    .invert_y = DT_INST_PROP(0, invert_y),
+    .sleep_en = DT_INST_PROP(0, sleep),
+    .no_taps = DT_INST_PROP(0, no_taps),
+#ifdef CONFIG_PINNACLE_TRIGGER
+    .dr = GPIO_DT_SPEC_GET(DT_DRV_INST(0), dr_gpios),
+#endif
+};
+
+DEVICE_DT_INST_DEFINE(0, pinnacle_init, device_pm_control_nop, &pinnacle_data, &pinnacle_config, POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY, &pinnacle_driver_api);
```

Comment:
Do we want to assume from the get go that we only want to support one instance of this driver?

Seems like a bit of a risk.